Samples/Notifications/Push/cpp-console-unpackaged/cpp-console-unpackaged.cpp

```diff
@@ -0,0 +1,160 @@
+    {
+        auto channel{ result.Channel() };
+
+        std::cout << "\nWNS Channel URI: " << winrt::to_string(channel.Uri().ToString()) << std::endl;
+
+        // It's the caller's responsibility to keep the channel alive
+        co_return channel;
+    }
+    else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+    {
+        LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with the WNS Channel URI request!");
+        co_return nullptr;
+    }
+    else
+    {
+        LOG_HR_MSG(result.ExtendedError(), "Some other failure occurred.");
+        co_return nullptr;
+    }
+
+};
+
+winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel RequestChannel()
+{
+    auto task{ RequestChannelAsync() };
+    if (task.wait_for(std::chrono::minutes(5)) != AsyncStatus::Completed)
+    {
+        task.Cancel();
+        return nullptr;
+    }
+
+    auto result{ task.GetResults() };
+    return result;
+}
+
+// Subscribe to an event which will get signaled whenever a foreground notification arrives.
+void SubscribeForegroundEventHandler()
+{
+    winrt::event_token token{ PushNotificationManager::Default().PushReceived([](auto const&, PushNotificationReceivedEventArgs const& args)
+    {
+        auto payload{ args.Payload() };
+
+        std::string payloadString(payload.begin(), payload.end());
+        std::cout << "\nPush notification content received in the FOREGROUND: " << payloadString << std::endl;
+    }) };
+}
+
+int main()
+{
+    // Setup an event handler, so we can receive notifications in the foreground while the app is running.
+    SubscribeForegroundEventHandler();
+
+    PushNotificationManager::Default().Register();
+
+    auto args{ AppInstance::GetCurrent().GetActivatedEventArgs() };
+    switch (args.Kind())
+    {
+        // When it is launched normally (by the users, or from the debugger), the sample requests a WNS Channel URI and
+        // displays it, then waits for notifications. This user can take a copy of the WNS Channel URI and use it to send
+        // notifications to the sample
+        case ExtendedActivationKind::Launch:
+        {
+            // Request a WNS Channel URI which can be passed off to an external app to send notifications to.
+            // The WNS Channel URI uniquely identifies this app for this user and device.
+            PushNotificationChannel channel{ RequestChannel() };
+            if (!channel)
+            {
+                std::cout << "\nThere was an error obtaining the WNS Channel URI" << std::endl;
+
+                if (remoteId == winrt::guid { "00000000-0000-0000-0000-000000000000" })
+                {
+                    std::cout << "\nThe remoteId has not been set. Refer to the readme file accompanying this sample\nfor the instructions on how to obtain and setup a remote id" << std::endl;
+                }
+            }
+
+            std::cout << "\nPress 'Enter' at any time to exit App." << std::endl;
```

Comment:
\n

Reply:
my workaround is:

/etc/initramfs-tools/hooks/startdegradedraid

#!/bin/sh
#
# Copyright <C2><A9> 2006 Martin F. Krafft <email address hidden>
# based on the scripts in the initramfs-tools package.
# released under the terms of the Artistic Licence.
#
# $Id: hook 281 2006-12-08 08:14:44Z madduck $
#

set -eu

PREREQ="udev"

prereqs()
{
        echo "$PREREQ"
}

case ${1:-} in
  prereqs)
    prereqs
    exit 0
    ;;
esac

MDADM=$(command -v mdadm 2>/dev/null) || :
[ -x $MDADM ] || exit 0

DESTCONFIG=$DESTDIR/conf/md.conf

echo "MD_DEGRADED_ARGS=' '" >> $DESTCONFIG